File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
       int n = nums.length;
       k = k%n; //in case k is greater than n 
       int result[] = new int[n];//create a new array to store
       //iterate till the k 
       for(int i = 0;i<k;i++){
        result[i] =nums[n -k+i]; 
       }
       //copy first n-k elements to the end of result
       for(int i =0;i<n -k;i++){
        result[k+i] = nums[i];
       }
       //copy back to original array
       for(int i = 0;i<n;i++){
        nums[i] = result[i];
       }
    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        n = len(nums)
        k = k % n  # handle k > n

        result = [0] * n  # new array

        # copy last k elements to front
        for i in range(k):
            result[i] = nums[n - k + i]

        # copy first n-k elements after that
        for i in range(n - k):
            result[k + i] = nums[i]

        # copy back to original array
        for i in range(n):
            nums[i] = result[i]